src/backend/renderer/output/element/mod.rs

```diff
@@ -0,0 +1,443 @@
+use crate::{
+    backend::renderer::{ImportAll, Renderer},
+    utils::{Physical, Point, Rectangle, Scale},
+};
+
+mod id;
+
+pub use id::Id;
+use wayland_server::protocol::wl_buffer;
+
+pub mod surface;
+pub mod texture;
+
+/// The underlying storage for a element
+#[derive(Debug)]
+pub enum UnderlyingStorage<'a, R: Renderer> {
+    /// A wayland buffer
+    Wayland(wl_buffer::WlBuffer),
+    /// A texture
+    External(&'a R::TextureId),
+}
+
+/// A single render element
+pub trait RenderElement<R: Renderer> {
+    /// Get the unique id of this element
+    fn id(&self) -> &Id;
+    /// Get the current commit position of this element
+    fn current_commit(&self) -> usize;
```

Comment:
Is this a value which always increments or is it maxing out at 4 or so?

If it increments forever, assuming 144hz it would take 56 years to reach usize::MAX on 32-bit systems

Reply:
That depends on the implementor, the new `OutputRender` type just uses the value to feed it into `damage_since`.
The default impl for `damage_since` just checks if the commit is different and will then report the whole size as damaged.
But yeah, this is possibly an ever increasing value, in one implementation I just used `wrapping_add(1)`.
I want to add a type for damage tracked texture elements which would handle that internally, but I also want to change to usize to a custom type which increments interanally (probably with a `wrapping_add`.